Third party cookie check (#970)
SAP/luigi: docs/general-settings.md

  allowRules: ['microphone'],
  appLoadingIndicator: {
    hideAutomatically: true
  },

>   thirdPartyCookiesScriptLocation: 'someUrlWhereInit_htmlAndComplent_htmlAreHosted',

Reviewer: ```
  thirdPartyCookiesScriptLocation: 'someUrlWhereInit_htmlAndComplent_htmlAreHosted',
```

What does `Complent` mean? 🤔

```
* **allowRules** is an array of rules for the content in the iframe, managed by the HTML **allow** attribute. You can use one or more rules by adding them to the array, for example `allowRules: ['microphone', 'camera']`. Be aware that this mechanism requires the browser to support [Feature Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy).
* **appLoadingIndicator.hideAutomatically** allows you to disable automatic hiding of the app loading indicator, which is enabled by default in case the app loading indicator is being used. Take a look at the [App loading indicator](luigi-ux-features.md#app-loading-indicator) section on how to use this feature.

### Third-party cookies support check

> There are two possibilities to check the third party cookie support from user's browser.

Reviewer: ```suggestion
There are two ways to check whether the user's browser supports third-party cookies:
```

- **thirdPartyCookiesScriptLocation** is the URL to the page containing third-party cookies support check.

> To detect whether the user's browser supports the mechanism, use the script in the third-party-cookies catalog. Deploy these files on a domain different from your main application's and set thirdPartyCookiesScriptLocation to the init.html file. During initialization, Luigi detects the cookies support and produces an alert if cookies are disabled in the user's browser.

Reviewer: ```suggestion
To detect whether the user's browser supports the mechanism, use the script in the third-party cookies catalog. Deploy these files on a domain different from your main application's and set **thirdPartyCookiesScriptLocation** to the `init.html` file. During initialization, Luigi detects cookies support and produces an alert if cookies are disabled in the user's browser.
```

> * **thirdPartyCookieDetection** is a flag to check if third party cookies are supported. When set to `true`, Luigi detects during initialization of an external microfrontend the cookie support and produces an alert if cookies are disabled in the user's browsers.

Reviewer: ```suggestion
* **thirdPartyCookieDetection** is a flag to check if third-party cookies are supported. When set to `true`, Luigi detects cookie support during the initialization of an external micro frontend and produces an alert if cookies are disabled in the user's browsers.
```

```
* **allowRules** is an array of rules for the content in the iframe, managed by the HTML **allow** attribute. You can use one or more rules by adding them to the array, for example `allowRules: ['microphone', 'camera']`. Be aware that this mechanism requires the browser to support [Feature Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy).
* **appLoadingIndicator.hideAutomatically** allows you to disable automatic hiding of the app loading indicator, which is enabled by default in case the app loading indicator is being used. Take a look at the [App loading indicator](luigi-ux-features.md#app-loading-indicator) section on how to use this feature.

### Third-party cookies support check

There are two ways to check whether the user's browser supports third-party cookies:

> **thirdPartyCookieCheck** is an Object which expects a function called `thirdPartyCookieErrorHandling` and optional `thirdPartyCookiesScriptLocation` parameter. When `thirdPartyCookiesScriptLocation` is set the Luigi Core application checks the third party cookie support only once and not on every micro frontend call. If it is `not` set, the Luigi Core application checks the third party cookie support whenever a micro frontend will be loaded.

Reviewer: ```suggestion
**thirdPartyCookieCheck** is an object which expects a function called `thirdPartyCookieErrorHandling` and an optional **thirdPartyCookiesScriptLocation** parameter. When **thirdPartyCookiesScriptLocation** is set, the Luigi Core application checks third-party cookie support only once and not on every micro frontend call. If it is *not* set, the Luigi Core application checks third-party cookie support whenever a micro frontend is loaded.
```

>   * **thirdPartyCookieErrorHandling** is a function where you could call an alert like `Luigi.ux().showAlert({})`.

Reviewer: ```suggestion
  * `thirdPartyCookieErrorHandling` is a function where you could call an alert like `Luigi.ux().showAlert({})`.
```

```
* **allowRules** is an array of rules for the content in the iframe, managed by the HTML **allow** attribute. You can use one or more rules by adding them to the array, for example `allowRules: ['microphone', 'camera']`. Be aware that this mechanism requires the browser to support [Feature Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy).
* **appLoadingIndicator.hideAutomatically** allows you to disable automatic hiding of the app loading indicator, which is enabled by default in case the app loading indicator is being used. Take a look at the [App loading indicator](luigi-ux-features.md#app-loading-indicator) section on how to use this feature.

### Third-party cookies support check

There are two ways to check whether the user's browser supports third-party cookies:

> **thirdPartyCookieCheck** is an Object which expects a function called `thirdPartyCookieMessage` and ether `thirdPartyCookiesScriptLocation` or `thirdPartyCookieDetection`.

Reviewer: ```suggestion
**thirdPartyCookieCheck** is an Object which expects a function called `thirdPartyCookieMessage` and either the `thirdPartyCookiesScriptLocation` or `thirdPartyCookieDetection` parameter.
```

>   * **thirdPartyCookieMessage** as a function where you could call an alert like `Luigi.ux().showAlert({})`.

Reviewer: ```suggestion
  * **thirdPartyCookieMessage** is a function where you could call an alert like `Luigi.ux().showAlert({})`.
```
